src/java.desktop/share/classes/sun/awt/shell/ShellFolder.java

```diff
@@ -207,6 +207,15 @@ public Image getIcon(boolean getLargeIcon) {
         return null;
     }
 
+    /**
+     * Returns the icon of the specified size used to display this shell folder.
+     *
+     * @param size size of the icon > 0 (Valid range: 1 to 256)
```

Comment:
I'm unsure the valid range of 1 to 256 makes sense provided the icon smaller than 16×16 is never returned (on Windows at least).

Reply:
Well, user still can request 1x1 icon - we will return the multiresolution image with minimal (1x1) icon inside that will be scaled every time the actual painting occurs. The size will define the layout of the component with the icon and can be auto-generated from the user code so i do not see why we should limit the lowest requested size - especially in the shared instance that not only specific for Windows platform.

Reply:
Even though 1×1 icon doesn't make much sense, you're right imposing a limitation is no good either.

Reply:
Even though 1×1 icon doesn't make much sense, you're right imposing a limitation is no good either.